src/components/TodoList.jsx

```diff
@@ -13,51 +13,121 @@ class TodoListXP extends React.Component {
 
         this.state = {
             tasks: [
-                "do one thing",
-                "another one",
-                "last one to do"
+                { taskName: "do one thing", date: new Date() },
+                { taskName: "another one", date: new Date() },
+                { taskName: "last one to do", date: new Date() }
             ],
-            input: "nichego net"
+            input: ""
         };
-
         this.addTask = this.addTask.bind(this);
         this.removeTask = this.removeTask.bind(this);
         this.onInputChange = this.onInputChange.bind(this);
+        this.removeAllTasks = this.removeAllTasks.bind(this)
+        this.sortTasks = this.sortTasks.bind(this)
     }
 
-    removeTask(text) {
-        const { tasks } = this.state;
 
+    sortTasks(event) {
```

Comment:
This is sadly not always working properly.

Please think of how to fix this bug: 
If you select any type of sorting, and then add a new TODO in the list - it always adds in the end, ignoring sorting.
Let's fix this up.